### DISCUSSION TITLE
2.3.24

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#12086] [General]
*  WSL: Change VM creation to fall back to no networking instead of failing
*  Manually call IPlan9FileSystem::Teardown() to work around destructor hang in ni_release
*  Update Windows App SDK to 1.6

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/2.3.24'>2.3.24</a>.</em>

**Comment by DoctorWho8** [comment]
And what about the kernel version? Has that changed for the better?

**Comment by qlenlen** [comment]
Waiting for 6.6 come back.

**Comment by KylinDemons** [comment]
hope kernel 6.6 come back.

**Reply by MorphBonehunter** [reply]
[I don't think that there will be an update in the near future.](https://github.com/microsoft/WSL/issues/11884#issuecomment-2401327352)

**Comment by astroboylrx** [comment]
Whenever an update is released, Windows Store terminates my WSL and updates it for me, regardless of any running processes. I'm using WSL2 for work for a long time. I don't think the update was this aggressive previously, at least before 2024.
**Is there something I can do to prevent this? Thanks in advance.**

Maybe it is not Windows Store who actually did the update? Maybe the Store just shows an update was done? What module performs the update?
I already turned off auto app updates in the Store settings and even enabled "Turn off Automatic Downloads and Install of updates" in the group policy.

```
WSL version: 2.3.24.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.65
MSRDC version: 1.2.5620
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22635.4145
```

**Reply by ZupoLlask** [reply]
I totally agree WSL updates cannot be handled like this.

**Reply by mamhadu** [reply]
All updates, when no automatic update is turned on, show require user interaction. This is abusing privileges without event notifying the user of the update.

**Reply by kevinburkeland** [reply]
not a perfect workaround but my trick is to just run winget upgrade --all before starting on work each day, that way I get the updates out of the way before I am in the middle of something, but as with everything your milage may vary.

**Comment by Avliis** [comment]
This work for me
[wsl.2.3.24.0.x64.msi](https://github.com/microsoft/WSL/releases/download/2.3.24/wsl.2.3.24.0.x64.msi)
thank you so much for the help!

**Comment by jogiji** [comment]
cannot install getting this error 
`PS C:\Users\rajiv> Wsl --update --pre-release
Checking for updates.
Updating Windows Subsystem for Linux to version: 2.3.24.
The older version of Windows Subsystem for Linux cannot be removed.  Contact your technical support group.
Update failed (exit code: 1603).
Log file: C:\Users\rajiv\AppData\Local\Temp\wsl-install-logs.txt
Error code: Wsl/UpdatePackage/ERROR_INSTALL_FAILURE`

**Reply by Avliis** [reply]
[wsl.2.3.24.0.x64.msi](https://github.com/microsoft/WSL/releases/download/2.3.24/wsl.2.3.24.0.x64.msi)

**Reply by SdisRoy** [reply]
> [wsl.2.3.24.0.x64.msi](https://github.com/microsoft/WSL/releases/download/2.3.24/wsl.2.3.24.0.x64.msi)

Its working to me! i have suceffuly install the msi version manualy instread of the command line automatic version.
Thks

**Comment by Utojaus** [comment]
Docker can not launch : deploying WSL2 distributions
ensuring main distro is deployed: deploying "docker-desktop": importing WSL distro "WSL2 is not supported with your current machine configuration.\r\nPlease enable the \"Virtual Machine Platform\" optional component and ensure virtualization is enabled in the BIOS.\r\nEnable \"Virtual Machine Platform\" by running: wsl.exe --install --no-distribution\r\nFor information please visit https://aka.ms/enablevirtualization\r\nError code: Wsl/Service/RegisterDistro/CreateVm/HCS/HCS_E_HYPERV_NOT_INSTALLED\r\n" output="docker-desktop": exit code: 4294967295: running WSL command wsl.exe C:\WINDOWS\System32\wsl.exe --import docker-desktop <HOME>\AppData\Local\Docker\wsl\main C:\Program Files\Docker\Docker\resources\wsl\wsl-bootstrap.tar --version 2: WSL2 is not supported with your current machine configuration.
Please enable the "Virtual Machine Platform" optional component and ensure virtualization is enabled in the BIOS.
Enable "Virtual Machine Platform" by running: wsl.exe --install --no-distribution
For information please visit https://aka.ms/enablevirtualization
Error code: Wsl/Service/RegisterDistro/CreateVm/HCS/HCS_E_HYPERV_NOT_INSTALLED
: exit status 0xffffffff
checking if isocache exists: CreateFile \\wsl$\docker-desktop-data\isocache\: The network name cannot be found.
